### ISSUE TITLE
Examples are outdated

### ISSUE THREAD

**Issue by JLandersZen** [confluentinc/examples#153]
Throughout the consumer examples are references to deprecated methods and classes.  Seems these examples don't demonstrate the correct usage of these classes.  

|Description	| Resource	| Path	| Location	
|-------------|--------------|---------|--------------|
|The method addFetch(String, int, long, int) from the type FetchRequestBuilder is deprecated	| | SimpleConsumerExample.java	 | /consumer/src/main/java/io/confluent/examples/consumer	| line 147	
|The method build() from the type FetchRequestBuilder is deprecated	| SimpleConsumerExample.java	| /consumer/src/main/java/io/confluent/examples/consumer	| line 148
|The method clientId(String) from the type FetchRequestBuilder is deprecated	| SimpleConsumerExample.java	| /consumer/src/main/java/io/confluent/examples/consumer	| line 146
|The method close() from the type SimpleConsumer is deprecated	| SimpleConsumerExample.java	| /consumer/src/main/java/io/confluent/examples/consumer	| line 73
|The method close() from the type SimpleConsumer is deprecated	| SimpleConsumerExample.java	| /consumer/src/main/java/io/confluent/examples/consumer	| line 165
|The method close() from the type SimpleConsumer is deprecated	| SimpleConsumerExample.java	| /consumer/src/main/java/io/confluent/examples/consumer	| line 199
|The method createMessageStreams(Map<String,Integer>, Decoder<Object>, Decoder<Object>) from the type ConsumerConnector is deprecated	| ConsumerGroupExample.java	| /consumer/src/main/java/io/confluent/examples/consumer	| line 70
|The method createMessageStreams(Map<String,Integer>, Decoder<String>, Decoder<Object>) from the type ConsumerConnector is deprecated	| AvroClicksSessionizer.java	| /specific-avro-consumer/src/main/java/io/confluent/examples/consumer	| line 103
|The method fetch(FetchRequest) from the type SimpleConsumer is deprecated	| SimpleConsumerExample.java	| /consumer/src/main/java/io/confluent/examples/consumer	| line 149
|The method getOffsetsBefore(OffsetRequest) from the type SimpleConsumer is deprecated	| SimpleConsumerExample.java	| /consumer/src/main/java/io/confluent/examples/consumer	| line 88
|The method iterator() from the type KafkaStream is deprecated	| AvroClicksSessionizer.java	| /specific-avro-consumer/src/main/java/io/confluent/examples/consumer	| line 104
|The method iterator() from the type KafkaStream is deprecated	| ConsumerLogic.java	| /consumer/src/main/java/io/confluent/examples/consumer	| line 34
|The method next() from the type ConsumerIterator is deprecated	| AvroClicksSessionizer.java	| /specific-avro-consumer/src/main/java/io/confluent/examples/consumer	| line 109

**Comment by guozhangwang** [comment]
@JLandersZen starting in 3.3.x we have migrated the Kafka Streams examples to its own repo and have the online docs also pointing to that repo now: https://github.com/confluentinc/kafka-streams-examples

**Comment by JLandersZen** [comment]
@guozhangwang - none of these files are part of the kafka-streams examples.  These are the basic Kafka usage examples found in this repo.

**Comment by NishanthShajahan** [comment]
I was looking at  the specific Record example where  you just use the serializer and pass schema registry url in producer . Is the schema registration done automatically  by  avro serializer ? . Under  what name/subject will that schema  registered ?. If I want  to retrieve/compare the schema via schema registry  rest apis  wouldn't that be needed

**Comment by cricket007** [comment]
@NishanthShajahan 

> Is the schema registration done automatically by avro serializer ? 

Yes, [during serialize()](https://github.com/confluentinc/schema-registry/blob/master/avro-serializer/src/main/java/io/confluent/kafka/serializers/AbstractKafkaAvroSerializer.java#L79)

Under what name/subject will that schema registered ?

[`${topic}-key` and `${topic}-value`](https://github.com/confluentinc/schema-registry/blob/master/avro-serializer/src/main/java/io/confluent/kafka/serializers/subject/TopicNameStrategy.java#L35)

**Comment by jconlon** [comment]
Do these examples work with the latest 5.0.0 CPlatform?  If not where can I find a complete maven based Producer example?

**Comment by cricket007** [comment]
@jconlon Older clients are still capable of communicating with newer brokers. If you find a problem specifically using the code with 5.0, feel free to open a new issue.

**Comment by jconlon** [comment]
My fault - broker was started on another host from my client using  [Confluent Enterprise QuickStart Docker](https://docs.confluent.io/current/quickstart/ce-docker-quickstart.html) .  Could not connect to the broker with the <hostname>:9092.  Fixed it by adding an alias host name of `broker` in my client machine's /etc/hosts to that remote host and used: 
`props.put("bootstrap.servers", "broker:9092");`
